packages/gatsby-dev-cli/src/index.js

```diff
@@ -65,7 +61,20 @@ gatsby-dev --set-path-to-repo /path/to/my/cloned/version/gatsby
   process.exit()
 }
 
-if (!argv.packages && _.isEmpty(gatsbyPackages)) {
+const localPkg = JSON.parse(fs.readFileSync(`package.json`))
+let packages = Object.keys(
+  _.merge({}, localPkg.dependencies, localPkg.devDependencies)
+)
+
+if (argv.copyAll) {
+  packages = fs.readdirSync(path.join(gatsbyLocation, `packages`))
+} else {
+  const { dependencies } = JSON.parse(fs.readFileSync(path.join(gatsbyLocation, `packages/gatsby/package.json`)))
+  const extraGatsbyPackages = Object.keys(dependencies).filter(d => d !== `gatsby-plugin-page-creator`)
```

Comment:
why is `gatsby-plugin-page-creator` excluded? is there any problem with it?

Reply:
I got that impression from @KyleAMathews 🤷‍♂️ I may have mis-interpreted him though.

> The copy all thing works. We could/should actually just bring those packages (other than gatsby-page-creator) as they're not used outside of gatsby itself.

So yeah, will remove that check!

Reply:
Missed a bit of the sentence there 😅

I meant we should bring packages depended by gatsby that aren't used outside of gatsby inside the gatsby package as the separation isn't needed.

So the logic should be actually to _always_ copy packages depended on by gatsby (that we maintain).

Reply:
is it worth to try to do intersection of `extraGatsbyPackages` with packages we actually have in repository? right now extraGatsbyPackages will contain stuff like `lodash` and other upstream packages, correct?

Reply:
Two scenarios in this PR:

1. --copy-all: this is basically a naive/dumb (but effective) approach where every package in gatsby/ is copied to the current project; this will be most useful in e2e scenario
2. normal/default: Added the packages that the package gatsby directly depends upon (gatsby-link, gatsby-cli, gatsby-plugin-page-creator, etc.)

So I think we're good here, right?

Reply:
@pieh the later filter catches it! We presume every package starting with gatsby- is a "gatsby" package, which is reasonable, right?

I could make the algo smarter and check if the directory exists, but that'd be a bit slower probably 🤷‍♂️

Reply:
Yeah, was just clarifying my comment. This PR is great. Just suggesting possible future work